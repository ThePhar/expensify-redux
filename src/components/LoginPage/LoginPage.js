import React         from 'react';
import { connect }   from 'react-redux';
import { startLogin} from '../../actions/auth';

import './LoginPage.scss';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify Redux</h1>
            <p>Let's get your expenses under control.</p>
            <button onClick={startLogin}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);