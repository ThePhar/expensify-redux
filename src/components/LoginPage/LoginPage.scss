@import '../../index.scss';

// Colors
$transparent-white: fade-out(white, 0.05);

.box-layout {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100vw;
    height: 100vh;
    background: url('background.jpg');
    background-size: cover;

    .box-layout__box {
        width: 30rem;
        padding: $l-size $m-size;
        text-align: center;
        border-radius: 3px;
        background-color: $transparent-white;
    }

    .box-layout__title {
        line-height: 1;
        margin: 0 0 $m-size 0;
    }

    button {
        font-size: 1.8rem;
        font-weight: 300;
        padding: $s-size;
        color: white;
        border: none;
        border-radius: 3px;
        background-color: cornflowerblue;
    }
}