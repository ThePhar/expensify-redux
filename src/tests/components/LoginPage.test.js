import { shallow }   from 'enzyme';
import React         from 'react';
import { LoginPage } from '../../components/LoginPage/LoginPage';

let startLogin, wrapper;
beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin={startLogin} />)
});

test('Should render LoginPage.', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin when clicked.', () => {
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});