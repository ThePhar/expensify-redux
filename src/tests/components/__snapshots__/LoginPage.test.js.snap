// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage. 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify Redux
    </h1>
    <p>
      Let's get your expenses under control.
    </p>
    <button
      onClick={[MockFunction]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
